bpo-44859: Improve error handling in sqlite3 and change some errors (#27654)
python/cpython: Misc/NEWS.d/next/Library/2021-08-07-17-28-56.bpo-44859.CCopjk.rst

> Improve error handling in :mod:`sqlite3` and change some errors to more
> appropriate.

Reviewer: ```suggestion
Improve error handling in :mod:`sqlite3` and raise more accurate exceptions.
```

* :exc:`MemoryError` is now raised instead of :exc:`sqlite3.Warning` when memory is not enough for encoding a statement to UTF-8 in ``Connection.__call__()`` and ``Cursor.execute()``.
* :exc:`UnicodEncodeError` is now raised instead of :exc:`sqlite3.Warning` when the statement contains surrogate characters in ``Connection.__call__()`` and ``Cursor.execute()``.
* :exc:`TypeError` is now raised instead of :exc:`ValueError` for non-string script argument in ``Cursor.executescript()``.
* :exc:`ValueError` is now raised for script containing the null character instead of truncating it in ``Cursor.executescript()``.
* Correctly handle exceptions raised when getting boolean value of the result of the progress handler.

> * Add may tests covering different exceptional cases.

Reviewer: Maybe something like this?

```suggestion
* Add many tests covering different corner cases.
```
